scilpy/tracking/tools.py

```diff
@@ -147,28 +149,187 @@ def get_subset_streamlines(sft, max_streamlines, rng_seed=None):
+        return_indices_only.
     """
 
     rng = np.random.RandomState(rng_seed)
     ind = np.arange(len(sft.streamlines))
     rng.shuffle(ind)
 
-    subset_streamlines = list(np.asarray(sft.streamlines, dtype=object)[
-                              ind[:max_streamlines]])
-    subset_data_per_point = sft.data_per_point[ind[:max_streamlines]]
-    subset_data_per_streamline = sft.data_per_streamline[ind[:max_streamlines]]
+    if return_indices_only:
+        return ind[:max_streamlines]
 
-    subset_sft = StatefulTractogram.from_sft(
-        subset_streamlines, sft,
-        data_per_point=subset_data_per_point,
-        data_per_streamline=subset_data_per_streamline)
+    return sft[ind[:max_streamlines]]
 
-    return subset_sft
+
+def get_n_subsets_streamlines_sequentially(orig_sft, chunk_sizes):
+    """
+    Get n chunks of subsets of streamlines, sequentially through the sft.
+
+    Parameters
+    ----------
+    orig_sft: StatefulTractogram
+        Initial tractogram to subdivide
+    chunk_sizes: list[int]
+        Number of streamlines to keep per chunk.
+
+    Return
+    ------
+    all_chunks: list[StatefulTractogram]
+        The filtered streamlines as a sfts.
+    """
+    if sum(chunk_sizes) > len(orig_sft):
+        raise ValueError("You asked for more streamlines than are available.")
+
+    nb_chunks = len(chunk_sizes)
+
+    curr = 0
+    sfts = []
+    for i in range(nb_chunks):
+        nb_str = chunk_sizes[i]
+        sfts.append(orig_sft[curr:curr + nb_str])
+        curr += chunk_sizes[i]
+
+    return sfts
+
+
+def get_n_subsets_streamlines_random(orig_sft, chunk_sizes, seed,
+                                     return_indices_only=False):
+    """
+    Get n chunks of subsets of streamlines.
+
+    Parameters
+    ----------
+    orig_sft: StatefulTractogram
+        Initial tractogram to subdivide
+    chunk_sizes: list[int]
+        Number of streamlines to keep per chunk.
+    seed: int
+        Random seed.
+    return_indices_only: bool
+        If true, return a random list of indices. Else, return the Stateful
+        Tractogram containing the chosen streamlines.
+
+    Return
+    ------
+    all_chunks: list[list[int]] or list[StatefulTractogram]
+        The filtered streamlines as a sfts, or the lists of indices if
+        return_indices_only. Last item of the list contains streamlines that
+        were rejected.
+    """
+    if sum(chunk_sizes) > len(orig_sft):
+        raise ValueError("You asked for more streamlines than are available.")
+
+    nb_chunks = len(chunk_sizes)
+    final_indices = [[] for _ in chunk_sizes]
+    orig_indices_left = np.arange(len(orig_sft))
+    if return_indices_only:
+        sft_left = orig_sft  # No need to keep orig_sft intact.
+    else:
+        sft_left = orig_sft[:]  # Equivalent of deep copy
+
+    for i in range(nb_chunks):
+        nb_to_keep = chunk_sizes[i]
+
+        sub_ind = get_subset_streamlines(
```

Comment:
you could shuffle ids just once no? and create your chunks from these indices?

Reply:
Yes I could. I was just re-using the already-existing method.  I'll change that.